dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/JdbcEventStore.java

```diff
@@ -255,7 +251,9 @@ private List<Event> fetchEvents(EventQueryParams queryParams, PageParams pagePar
     if (pageParams == null) {
       eventsByUid = new HashMap<>();
     } else {
-      eventsByUid = new HashMap<>(pageParams.getPageSize());
+      eventsByUid =
+          new HashMap<>(
+              pageParams.getPageSize() + 1); // get extra event to determine if there is a nextPage
```

Comment:
if there's an extra event, how do you paginate them?

Reply:
See previous PR https://github.com/dhis2/dhis2-core/pull/20000

https://github.com/dhis2/dhis2-core/blob/3083743f6a3f38b62d339a6199631f3580a99294/dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/Page.java#L70-L73

Its the logic we have in https://github.com/dhis2/dhis2-core/blob/3083743f6a3f38b62d339a6199631f3580a99294/dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/HibernateEventChangeLogStore.java#L173